src/hypothesis/extra/datetime.py

```diff
@@ -123,3 +123,26 @@ def times(allow_naive=None, timezones=None):
 
 def datetime_to_time(dt):
     return dt.timetz()
+
+SECS_IN_DAY = 3600 * 24
+MICROS_IN_SEC = 1000000
+MICROS_IN_DAY = MICROS_IN_SEC * SECS_IN_DAY
+class TimedeltaStrategy(SearchStrategy):
+
+    def __init__(self, min_value=dt.timedelta.min, max_value=dt.timedelta.max):
+        self.max_micros = max_value.microseconds + (max_value.seconds + max_value.days * SECS_IN_DAY) * MICROS_IN_SEC
+        self.min_micros = min_value.microseconds + (min_value.seconds + min_value.days * SECS_IN_DAY) * MICROS_IN_SEC
+
+    def do_draw(self, data):
+        new_td_micros = cu.integer_range(data, self.min_micros, self.max_micros)
+
+        days = new_td_micros / MICROS_IN_DAY
+        secs = (new_td_micros - days * MICROS_IN_DAY) / MICROS_IN_SEC
+        micros = (new_td_micros - days * MICROS_IN_DAY - secs * MICROS_IN_SEC)
+
+        return dt.timedelta(days=days, seconds=secs, microseconds=micros)
```

Comment:
You can just pass the `microseconds` value to `timedelta` directly, and let it handle the day/second conversion. For example:

```pycon
>>> from datetime import timedelta
>>> t = timedelta(microseconds=1234567890987654321)
>>> (t.days, t.seconds, t.microseconds)
(14288980, 18987, 654321)
```